### ISSUE TITLE
IllegalStateException if IP is stored as an attribute

### ISSUE THREAD

**Issue by MadhuNEC** [telefonicaid/fiware-cygnus#2226]
I have setup Orion with Cygnus(1.15.0) and STH-Comet.
Now, I want to store IP as an attribute. I have created below subscription and then created an entity over Orion.

**Create a Subscription on Orion with below payload:**
```
{
  "id": "1234abcd",
  "description": "Subscriptions for noticing Sensor",
  "expires": "2024-01-01T00:00:00.00Z",
  "status": "active",
  "subject": {
    "entities": [
      {
        "idPattern": "test.*",
        "type": "Sensor" 
      }
    ],
    "condition": {
      "attrs": []
    }
  },
  "notification": {
    "attrs": [],
    "attrsFormat": "legacy",
    "http": {
      "url": "http://cygnus:5050/notify" 
    }
  }
}
```

**Now, Create an entity on Orion with below payload:**
```
{
  "id": "test.1",
  "type": "Sensor",
  "ipAddress": {
    "type": "Text",
    "value": "192.168.56.101",
    "metadata": {}
  },
  "uploadDate": {
    "type": "DateTime",
    "value": "2023-04-02T04:21:20.00Z",
    "metadata": {}
  }
}
```

**Error:**
```
 time=2023-04-02T04:56:01.515Z | lvl=ERROR | corr=7c2b17bc-cb92-11ed-b5f0-0242c0a85005 | trans=426b6859-1cd9-432a-a002-dd4a06df5fa4 | srv=default | subsrv=/ | comp=cygnus-ngsi | op=run | msg=org.apache.flume.SinkRunner$PollingRunner[158] : Unable to deliver event. Exception follows.
 java.lang.IllegalStateException: close() called when transaction is OPEN - you must either commit or rollback first
       at com.google.common.base.Preconditions.checkState(Preconditions.java:172)
       at org.apache.flume.channel.BasicTransactionSemantics.close(BasicTransactionSemantics.java:179)
       at com.telefonica.iot.cygnus.sinks.NGSISink.processNewBatches(NGSISink.java:599)
       at com.telefonica.iot.cygnus.sinks.NGSISink.process(NGSISink.java:371)
       at org.apache.flume.sink.DefaultSinkProcessor.process(DefaultSinkProcessor.java:67)
       at org.apache.flume.SinkRunner$PollingRunner.run(SinkRunner.java:145)
       at java.lang.Thread.run(Thread.java:748)
```
	   
**Analysis:**
I have checked the cygnus code. Below condition results into true as `isANumber()` returns true.
https://github.com/telefonicaid/fiware-cygnus/blob/master/cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSISTHSink.java#L341=
But below `update()` fails as it cannot recognize IP(192.168.56.101) as a valid Double.
https://github.com/telefonicaid/fiware-cygnus/blob/master/cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSISTHSink.java#L349

**Comment by MadhuNEC** [comment]
Hi @fgalan ,

Could you please provide your feedback in this issue?

Thanks!

**Comment by fgalan** [comment]
Cygnus 1.15.0 is an ancient version (released more than 3 years ago :)

Could you test if the same issue happens with newests version (Cygnus 2.20.0) and tell us about, please?

**Comment by MadhuNEC** [comment]
Hi @fgalan ,

Thanks for you support!

I have checked for latest version also. Same issue is persisting on latest version too. Below error is found in logs and no collection is created for aggregated entities:
`time=2023-04-04T13:20:57.361Z | lvl=ERROR | corr=N/A | trans=N/A | srv=N/A | subsrv=N/A | comp=cygnus-ngsi | op=processRollbackedBatches | msg=com.telefonica.iot.cygnus.sinks.NGSISink[414] : multiple pointsStack trace: [sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1914), sun.misc.FloatingDecimal.parseDouble(FloatingDecimal.java:110), java.lang.Double.parseDouble(Double.java:538), java.lang.Double.<init>(Double.java:608), com.telefonica.iot.cygnus.sinks.NGSISTHSink$STHAggregator.aggregate(NGSISTHSink.java:349), com.telefonica.iot.cygnus.sinks.NGSISTHSink.persistBatch(NGSISTHSink.java:104), com.telefonica.iot.cygnus.sinks.NGSISink.processRollbackedBatches(NGSISink.java:398), com.telefonica.iot.cygnus.sinks.NGSISink.process(NGSISink.java:370), org.apache.flume.sink.DefaultSinkProcessor.process(DefaultSinkProcessor.java:67), org.apache.flume.SinkRunner$PollingRunner.run(SinkRunner.java:145), java.lang.Thread.run(Thread.java:750)] Sink: sth-sink Destination: test_/_test.1_Sensor
`

**Comment by fgalan** [comment]
Fixed by PR #2228
